finesse/gui/stepper_motor_view.py

```diff
@@ -51,7 +63,25 @@ def _add_checkable_button(self, name: str) -> QPushButton:
     def _preset_clicked(self, btn: QPushButton) -> None:
         """Move the stepper motor to preset position."""
         # If the motor is already moving, stop it now
+        pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.notify_on_stopped")
         pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.stop")
 
         target = float(self.angle.value()) if btn is self.goto else btn.text().lower()
         pub.sendMessage(f"device.{STEPPER_MOTOR_TOPIC}.move.begin", target=target)
+
+    def _update_mirror_position_display(self, moved_to: float | None) -> None:
+        """Display the angle the mirror has moved to.
+
+        moved_to will be None if mirror is moving, in which case display "Moving...",
+        otherwise, display angle or its associated name.
+        """
+        if moved_to is None:
+            self.mirror_position_display.setText("Moving...")
+        else:
+            if moved_to in ANGLE_PRESETS.values():
+                preset = list(ANGLE_PRESETS.keys())[
+                    list(ANGLE_PRESETS.values()).index(moved_to)
+                ]
+                self.mirror_position_display.setText(preset.upper())
+            else:
+                self.mirror_position_display.setText(f"{moved_to}°")
```

Comment:
I'd be inclined to use a generator expression here, e.g.:

```suggestion
            preset = next((k for k, v in ANGLE_PRESETS.items() if v == moved_to), None)
            if preset:
                self.mirror_position_display.setText(preset.upper())
            else:
                self.mirror_position_display.setText(f"{moved_to}°")
```

Reply:
Thanks! I knew there had to be a more elegant way haha